heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -209,6 +209,31 @@ public Boolean deleteSchedulerLocation(String topologyName) {
   }
 
   /**
+   * Clean all states of a heron topology. This goes through each piece of state that needs
+   * to be cleaned up and will log out a warning if it could not be cleaned up properly.
+   * TMasterLocation, PackingPlan, PhysicalPlan, SchedulerLocation, ExecutionState, and Topology
+   * @param topologyName the name of the topology that we should clean the state for
+   */
+  public void cleanState(String topologyName) {
```

Comment:
Actually, we could add delete to the enum so it really won't get missed:
```
    public void delete(IStateManager stateManager, String topology) {
      switch(this) {
        case TMASTER_LOCATION: stateManager.deleteTMasterLocation(topology);
        case ....
      }
    }
```

then we could do things like this:
```
for (StateLocation location : StateLocation.values()) {
  location.delete(stateManager, topologyName)
}
```